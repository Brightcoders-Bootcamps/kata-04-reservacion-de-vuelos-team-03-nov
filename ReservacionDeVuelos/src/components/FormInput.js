import React from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  Image,
  Pressable,
  TouchableOpacity,
} from 'react-native';
import {windowHeight, windowWith} from '../utils/Dimensions';
import colors from '../utils/Colors';

const FormInput = ({
  placeholderText,
  labelValue,
  secureTextEntry,
  source,
  ...rest
}) => {
  let ruta = source;

  return (
    <>
      <View style={styles.inputContainer}>
        <TextInput
          value={labelValue}
          style={styles.input}
          placeholder={placeholderText}
          secureTextEntry={secureTextEntry}
        />
      </View>
      {ruta != undefined && (
        <TouchableOpacity style={styles.viewIconGoogle}>
          <Image style={styles.iconGoogle} source={ruta} />
        </TouchableOpacity>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  inputContainer: {
    marginTop: 5,
    width: 332,
    marginBottom: 10,
    height: windowHeight / 15,
    borderColor: colors.black,
    borderRadius: 3,
    borderWidth: 1,
    alignItems: 'center',
    backgroundColor: colors.white,
  },
  input: {
    padding: 10,
    flex: 1,
    fontSize: 16,
    color: '#333',
    justifyContent: 'flex-start',
    alignItems: 'center',
    borderColor: colors.black,
  },
  viewIconGoogle: {
    width: '20%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  iconGoogle: {
    height: 30,
    width: 30,
    marginLeft: 280,
    marginTop: -60,
  },
});

export default FormInput;
