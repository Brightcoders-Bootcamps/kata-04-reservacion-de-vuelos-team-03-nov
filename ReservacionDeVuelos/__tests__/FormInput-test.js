import React from 'react';
import renderer from 'react-test-renderer';
import FormInput from '../src/components/FormInput';

test('renders correctly', () => {
  const tree = renderer.create(<FormInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
