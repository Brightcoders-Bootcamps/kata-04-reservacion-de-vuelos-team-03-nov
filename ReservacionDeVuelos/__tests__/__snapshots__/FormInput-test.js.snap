// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "height": 88.93333333333334,
        "marginBottom": 2,
        "marginLeft": 7,
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#000",
          "borderLeftWidth": 1,
          "borderTopWidth": 1,
          "color": "#000",
          "fontSize": 16,
          "height": 88.93333333333334,
          "marginBottom": 2,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
        "height": 88.93333333333334,
        "justifyContent": "center",
        "width": "15%",
      }
    }
  />
</View>
`;
